extensions/ql-vscode/src/data-extensions-editor/auto-modeler.ts

```diff
@@ -0,0 +1,171 @@
+import { ExternalApiUsage, MethodSignature } from "./external-api-usage";
+import { ModeledMethod } from "./modeled-method";
+import { extLogger } from "../common/logging/vscode";
+import { load as loadYaml } from "js-yaml";
+import { ProgressCallback, withProgress } from "../common/vscode/progress";
+import { createAutoModelV2Request, getCandidates } from "./auto-model-v2";
+import { runAutoModelQueries } from "./auto-model-codeml-queries";
+import { loadDataExtensionYaml } from "./yaml";
+import { ModelRequest, ModelResponse, autoModelV2 } from "./auto-model-api-v2";
+import { RequestError } from "@octokit/request-error";
+import { showAndLogExceptionWithTelemetry } from "../common/logging";
+import { redactableError } from "../common/errors";
+import { App } from "../common/app";
+import { CodeQLCliServer } from "../codeql-cli/cli";
+import { QueryRunner } from "../query-server";
+import { DatabaseItem } from "../databases/local-databases";
+import { Mode } from "./shared/mode";
+
+export class AutoModeler {
```

Comment:
Should we add some comments to the methods in here?

Reply:
Sure. Did you have anything specific in mind that you wanted commented?

Reply:
I might just do that in a separate PR when there's a bit more beef to it. But please let me know if you'd like anything specific commented :)